packages/fx-core/src/plugins/resource/apim/config.ts

```diff
@@ -266,17 +257,21 @@ export class SolutionConfig implements ISolutionConfig {
 }
 
 function checkAndGetOtherPluginConfig(
-  configOfOtherPlugins: ReadonlySolutionConfig,
+  pluginConfig: ReadonlyPluginConfig | Json,
   component: TeamsToolkitComponent,
   key: string,
   envName: string
 ): string {
-  const pluginConfig = configOfOtherPlugins.get(component);
-  if (!pluginConfig) {
-    throw BuildError(NoPluginConfig, component, ComponentRetryOperations[component]);
+  const value = pluginConfig.get ? pluginConfig.get(key) : (pluginConfig as Json)[key];
```

Comment:
Can we keep it same to the old logic? get the plugin config first then get the value?